### ISSUE TITLE
List the C dependencies

### ISSUE THREAD

**Issue by multimeric** [januarharianto/respR#143]
Hi, I had a lot of trouble installing this package. The dependencies of your dependencies require a whole collection of C libraries that you have to install manually. I actually wasn't able to get it working using only `apt` packages, but in the end the following `conda` packages worked for me:
```bash
conda install -c conda-forge libgdal udunits2
```

I just thought it might be helpful to document this somewhere, as well as ideally the packages needed for other environments like `apt`-systems (Ubuntu etc), probably also `brew` systems too.

Thanks for the package.

**Comment by nicholascarey** [comment]
Hi Michael. 

Thanks for using!

Do you know which ones in particular? We've never come across this problem before. Can you explain more why the dependencies-of-dependencies wouldn't get installed as with our dependencies on your system? 

I have to confess I know little about Apt or Linux in general. We do automated build testing on Linux/Ubuntu (e.g. [here](https://github.com/januarharianto/respR/actions/runs/1002234215)), at least in the `develop` branch where we are working on v2. 

Admittedly, it's been so long since we committed anything to `master` there may be build and install problems we are not aware of.

By the way, if you are feeling adventurous you could install the `develop` branch. This is a massive update, and should be out in a couple of weeks. It's pretty stable. Lots of changes, including code braking ones though. Only real issue at the moment is `auto_rate` is a little broken.

**Comment by multimeric** [comment]
Okay so when I run the suggested installation commands:
```R
install.packages("devtools")
devtools::install_github("januarharianto/respR")
```

I get:
```
ERROR: dependency ‘sf’ is not available for package ‘oce’
* removing ‘/home/michael/R/x86_64-pc-linux-gnu-library/3.6/oce’
ERROR: dependency ‘oce’ is not available for package ‘seacarb’
* removing ‘/home/michael/R/x86_64-pc-linux-gnu-library/3.6/seacarb’
ERROR: dependency ‘seacarb’ is not available for package ‘marelac’
* removing ‘/home/michael/R/x86_64-pc-linux-gnu-library/3.6/marelac’
[...]
Installing package into ‘/home/michael/R/x86_64-pc-linux-gnu-library/3.6’
(as ‘lib’ is unspecified)
ERROR: dependency ‘marelac’ is not available for package ‘respR’
* removing ‘/home/michael/R/x86_64-pc-linux-gnu-library/3.6/respR’
Warning messages:
1: In i.p(...) : installation of package ‘units’ had non-zero exit status
2: In i.p(...) : installation of package ‘sf’ had non-zero exit status
3: In i.p(...) : installation of package ‘oce’ had non-zero exit status
4: In i.p(...) :
  installation of package ‘seacarb’ had non-zero exit status
5: In i.p(...) :
  installation of package ‘marelac’ had non-zero exit status
6: In i.p(...) :
  installation of package ‘/tmp/RtmpkNyXeb/filee8ca665497e23/respR_1.1.0.0.tar.gz’ had non-zero exit status
```

I can't find them now, but the error log also includes a reason why the compilation of these packages fails, and it's due to missing c libraries (libgdal and udunits2 at least). 

This has happened to me on Ubuntu 20.04 and 20.10. I get the impression this happens because, as explained [here](https://github.com/dankelley/oce/issues/1717#issuecomment-671133742), there are no pre-compiled packages for Linux, so you have to have the system libraries already installed. Not sure why R works like that, though, considering Python manages to have binary packages for Linux without much trouble.

**Comment by nicholascarey** [comment]
Ok, i think this is more a limitation/weirdness of R than anything to do with us, and i think you will see it installing other packages. I don't know how other environments work, but I think it's related to the issues discussed [here](https://community.rstudio.com/t/meaning-of-common-message-when-install-a-package-there-are-binary-versions-available-but-the-source-versions-are-later/2431). I see this quite a lot when updating packages and R ask if you want to install the version that needs to be compiled, but then a compiled version will become available a few days later.

`sf` had version 1.0.1 released about a [week ago](https://cloud.r-project.org/web/packages/sf/index.html), and I guess has not yet got a compiled version available for Ubuntu, which causes a cascade of dependency errors. 

I don't know how to avoid this. I'm sure theres a way or setting to automatically ignore uncompiled versions and install the latest compiled one, or automatically compile it, but in my brief googling didn't find anything obvious.

**Comment by nicholascarey** [comment]
Yes - `sf 1.0.1` is only available uncompiled on my system (Mac) too. You can try installing the previous version with this

`devtools::install_version('sf', '1.0-0')`

check versions here 

https://packagemanager.rstudio.com/client/#/repos/1/packages/sf

Meanwhile, i just uninstalled `sf` it to check, tried this and now have the same issue  😭

**Comment by multimeric** [comment]
Even running that command hits this issue. I don't think `sf` has binary packages, or at least not for Linux.

**Comment by nicholascarey** [comment]
Well, I'm also still waiting on a Apple Silicon Mac one for `sf 1.0.1` too. I don't know why there isn't one. As i said it can take several days for compiled binaries to show up, but why this one has not got one yet after nearly 2 weeks i don't know. 

As i said, our Ubuntu [builds](https://github.com/januarharianto/respR/actions/runs/1002234215) are working fine. So they must be set up to do something different than your system, perhaps compiling them. You could try looking at the log to try and figure out what that is.

**Comment by multimeric** [comment]
Okay I've looked more into this, and I'm a bit confused. You have an explicit step in your Ubuntu builds that calculates and installs the Ubuntu library dependencies. This is exactly what I'm talking about:

https://github.com/januarharianto/respR/blob/64f19218d0ced7799e11c8a68bded7bf6370e376/.github/workflows/check-standard.yaml#L60-L66

I went and ran this command myself, and it produces the following output:
```
apt-get install -y libicu-dev
apt-get install -y zlib1g-dev
apt-get install -y libudunits2-dev
apt-get install -y libgdal-dev
apt-get install -y gdal-bin
apt-get install -y libgeos-dev
apt-get install -y libproj-dev
apt-get install -y make
apt-get install -y libxml2-dev
apt-get install -y pandoc
apt-get install -y libglpk-dev
apt-get install -y libgmp3-dev
```

So in other words, these *are* the apt dependencies that I was asking for in the first place. What I'm suggesting is to either list these in the installation page, or to point people to the `remotes::system_requirements` command, because this also tells you what to install and I don't think most people (myself included) are aware of its existence.
